### ISSUE TITLE
RCS1205 does not trigger which method has optional arguments

### ISSUE THREAD

**Issue by fgimian** [JosefPihrt/Roslynator#935]
**Product and Version Used**: 4.1.1

**Steps to Reproduce**:

```csharp
internal static class ExampleClass
{
    internal static void Example(string a, string b, string c, string? d = null)
    {
    }

    internal static void RunExample()
    {
        // Should trigger RCS1205 below because the provided arguments are out of order
        Example(c: "c", a: "a", b: "b");
        
        // Triggers RCS1205 only when all arguments are provided when method has optional args
        Example(c: "c", a: "a", b: "b", d: "d");
    }
}
```

**Actual Behavior**:

See above

**Expected Behavior**:

See above

Thanks heaps in advance!
Fotis
